libcache/cache.c

```diff
@@ -645,3 +645,49 @@ int cache_invalidate(cachectx_t *cache, const uint64_t begAddr, const uint64_t e
 
 	return EOK;
 }
+
+
+int cache_clean(cachectx_t *cache, const uint64_t begAddr, const uint64_t endAddr)
+{
+	int ret = EOK;
+	uint64_t addr = 0, end = endAddr, tag = 0, index = 0, begOffset = 0;
+	ssize_t writeCount = 0;
+	cacheline_t *linePtr = NULL;
+
+	if (begAddr > endAddr || begAddr > cache->srcMemSize) {
+		return -EINVAL;
+	}
+	if (begAddr < cache->srcMemSize && endAddr > cache->srcMemSize) {
+		end = cache->srcMemSize;
+	}
+
+	begOffset = cache_computeOffset(cache, begAddr);
+	addr = begAddr - begOffset;
+
+	mutexLock(cache->lock);
+
+	while (addr < end) {
+		index = cache_computeSetIndex(cache, addr);
+		tag = cache_computeTag(cache, addr);
+
+		linePtr = cache_findLine(&cache->sets[index], tag, LIBCACHE_TIMESTAMPS_NO_UPDATE);
```

Comment:
All this stuff is becoming more duplicated between various functions. Not in scope of this PR, but it might be good to think about refactorization